src/v/config/property.h

```diff
@@ -123,10 +131,180 @@ class property : public base_property {
     };
 };
 
+template<typename T>
+requires((std::is_integral_v<T>)&&(
+  !std::is_same_v<T, const bool>)) std::string_view property_type_name(T*) {
+    return "integer";
+}
+
+template<typename T>
+requires std::is_floating_point_v<T> std::string_view property_type_name(T*) {
+    return "number";
+}
+
+inline std::string_view property_type_name(const bool*) { return "boolean"; }
+
+inline std::string_view property_type_name(const std::chrono::milliseconds*) {
+    return "integer";
+}
+
+inline std::string_view property_type_name(const std::chrono::seconds*) {
+    return "integer";
+}
+
+inline std::string_view property_type_name(const ss::sstring*) {
+    return "string";
+}
+
+inline std::string_view property_type_name(const model::broker_endpoint*) {
+    return "broker_endpoint";
+}
+
+inline std::string_view property_type_name(const config::endpoint_tls_config*) {
+    return "endpoint_tls_config";
+}
+
+inline std::string_view property_type_name(const config::tls_config*) {
+    return "tls_config";
+}
+
+inline std::string_view property_type_name(const unresolved_address*) {
+    return "unresolved_address";
+}
+
+inline std::string_view
+property_type_name(const model::violation_recovery_policy*) {
+    return "string";
+}
+
+inline std::string_view
+property_type_name(const model::cleanup_policy_bitflags*) {
+    return "string";
+}
+
+inline std::string_view property_type_name(const model::timestamp_type*) {
+    return "string";
```

Comment:
integer?

Reply:
This is an enum that gets serialized to & from a string.

Reply:
ahh I see